cogs/monitor.py

```diff
@@ -13,96 +23,238 @@ async def on_message(self, message: discord.Message):
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        await available_subcommands(ctx)
 
     @monitor.group()
+    @admin_command()
     async def channel(self, ctx: commands.Context):
         """
         Commands for setting up channels to log problematic messages to.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        await available_subcommands(ctx)
 
-    @channel.command()
-    async def user(self, ctx: commands.Context):
+    @channel.command(name='user')
+    @admin_command()
+    async def channel_user(self, ctx: commands.Context, channel: Optional[discord.TextChannel] = None):
         """
         Set logging channel for monitored users.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        values = dict(monitor_user_log_id = channel.id if channel else None)
+        await GuildData.update_or_create(values, guild_id=ctx.guild.id)
+        get_guild_data.cache_clear()
+
+        if channel:
+            await ctx.send(f'Log channel for monitored users set to <#{channel.id}> for this guild.')
+        else:
+            await ctx.send('Log channel for monitored users removed for this guild.')
     
-    @channel.command()
-    async def message(self, ctx: commands.Context):
+    @channel.command(name='message')
+    @admin_command()
+    async def channel_message(self, ctx: commands.Context, channel: Optional[discord.TextChannel] = None):
         """
         Set logging channel for monitored message patterns.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        values = dict(monitor_message_log_id = channel.id if channel else None)
+        await GuildData.update_or_create(values, guild_id=ctx.guild.id)
+        get_guild_data.cache_clear()
 
+        if channel:
+            await ctx.send(f'Log channel for monitored messages set to <#{channel.id}> for this guild.')
+        else:
+            await ctx.send('Log channel for monitored messages removed for this guild.')
 
     @monitor.command()
     async def cleanup(self, ctx: commands.Context):
         """
         Remove users that are no longer in the server.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        monitor_users = await get_all_monitor_users()
+        users_removed = 0
+
+        for monitor_user in monitor_users:
+            if not ctx.guild.get_member(monitor_user.user_id):
+                await monitor_user.delete()
+                users_removed += 1
+
+        get_all_monitor_users.cache_clear()
+        
+        logging.debug(f"{users_removed} user(s) were removed from monitor.")
+        await ctx.send(f"{users_removed} user(s) were removed from monitor.")
     
     @monitor.command()
     async def list(self, ctx: commands.Context):
```

Comment:
Not sure what would be the best way to go about it, but maybe having a way of juts viewing messages and just viewing users would be good.